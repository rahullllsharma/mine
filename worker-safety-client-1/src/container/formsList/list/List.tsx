import type { LoaderParams as ListInfiniteLoaderParams } from "./components/loader/Loader";
import type { Form } from "../../../types/formsList/formsList";
import { useIsFirstRender } from "usehooks-ts";
import FormsTable from "../../table/FormsTable";
import { Loader as ListInfiniteLoader } from "./components/loader/Loader";
import { FormItems } from "./components/formItems/FormItems";

type ListState = "loading" | "complete";

type ListParams = {
  formsData: Form[];
  state?: ListState;
  disableFetchMore: boolean;
} & Partial<Pick<ListInfiniteLoaderParams, "onLoadMore">>;

function List({ formsData, state, onLoadMore, disableFetchMore }: ListParams) {
  const isFirstRender = useIsFirstRender();
  const isLoading = state === "loading";
  const isComplete = state === "complete";
  const hasInfiniteScroll = !!(onLoadMore && !isComplete && !disableFetchMore);

  return (
    <div className="overflow-y-auto">
      {/* desktop */}
      <div className="hidden lg:flex">
        <FormsTable formsData={formsData} />
      </div>
      {/* mobile */}
      <div className="lg:hidden grid gap-4 grid-cols-auto-fill-list-card">
        <FormItems formsData={formsData} />
      </div>
      {/* Include infinite scroll support */}
      {hasInfiniteScroll && (
        <ListInfiniteLoader
          onLoadMore={onLoadMore}
          isLoading={isLoading}
          shouldLoadMore={!isFirstRender}
        />
      )}
    </div>
  );
}

export { List };
export type { ListState, ListParams };
