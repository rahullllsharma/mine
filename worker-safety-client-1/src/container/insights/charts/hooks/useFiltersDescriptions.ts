import type { Contractor } from "@/types/project/Contractor";
import type { LibraryDivision } from "@/types/project/LibraryDivision";
import type { LibraryRegion } from "@/types/project/LibraryRegion";
import type { ProjectStatusOption } from "@/types/project/ProjectStatus";
import type { FilterProject } from "../../utils";
import type { ProjectFiltersPayload } from "../../projectFilters/ProjectFilters";
import type { PortfolioFiltersPayload } from "../../portfolioFilters/PortfolioFilters";
import type { FiltersPayload } from "../types";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { InsightsTab } from "../types";

type GenericDictionary = { id: string; name: string }[];

type GetPortfolioFiltersArgs = Required<
  Pick<FiltersDescriptionsParams, "descriptions">
> & {
  filters: PortfolioFiltersPayload;
};

type GetProjectFiltersArgs = Required<
  Pick<FiltersDescriptionsParams, "descriptions">
> & {
  filters: ProjectFiltersPayload;
};

export type FiltersDescriptionsParams = {
  filters?: FiltersPayload;
  tab: InsightsTab;
  descriptions: Partial<{
    projects: FilterProject[];
    regions: LibraryRegion[];
    divisions: LibraryDivision[];
    contractors: Contractor[];
    projectStatusOptions: ProjectStatusOption;
  }>;
};

export type FiltersDescriptionsReturn = {
  [key: string]: string | number | undefined;
}[];

/** Extract ONLY the selected filters names by the selected ids. */
const extractNamesBySelectedFilters = (
  filteredIds: string[] = [],
  originals: GenericDictionary = []
) =>
  originals
    .reduce<string[]>((acc, { id, name }) => {
      if (filteredIds.includes(id)) {
        acc.push(name);
      }
      return acc;
    }, [])
    .join(", ");

/** Extract and format the filters for a PORTFOLIO-type chart */
const getPortfolioFilters = (data: GetPortfolioFiltersArgs) => {
  const {
    workPackage: {
      attributes: {
        name,
        primeContractor,
        division,
        region,
        status,
        startDate: startDateAttr,
        endDate: endDateAttr,
      },
    },
  } = useTenantStore.getState().getAllEntities();

  const {
    projectStatuses,
    projectIds,
    regionIds,
    divisionIds,
    contractorIds,
    startDate,
    endDate,
  } = data.filters;

  const { projectStatusOptions, projects, regions, divisions, contractors } =
    data.descriptions;

  const portfolioFilters = [
    [
      status.label,
      {
        filters: projectStatuses,
        descriptions: projectStatusOptions as unknown as GenericDictionary,
      },
    ],
    [
      name.labelPlural,
      {
        filters: projectIds,
        descriptions: projects,
      },
    ],
    [
      region.labelPlural,
      {
        filters: regionIds,
        descriptions: regions,
      },
    ],
    [
      division.labelPlural,
      {
        filters: divisionIds,
        descriptions: divisions,
      },
    ],
    [
      primeContractor.labelPlural,
      {
        filters: contractorIds,
        descriptions: contractors,
      },
    ],
  ] as const;

  // Loops and filters only the selected filters from a list of all available filters.
  const availableFilters = portfolioFilters.reduce<Record<string, string>>(
    (acc, [filter, { filters, descriptions }]) => {
      if (Array.isArray(filters) && filters.length > 0) {
        acc[filter] = extractNamesBySelectedFilters(filters, descriptions);
      }

      return acc;
    },
    {}
  );

  return [
    {
      [startDateAttr.label]: startDate,
      [endDateAttr.label]: endDate,
      ...availableFilters,
    },
  ];
};

/** Extract and format the filters for a PROJECT-type chart */
const getProjectFilters = ({
  descriptions,
  filters,
}: GetProjectFiltersArgs) => {
  const {
    workPackage: {
      attributes: {
        name: workPackageName,
        externalKey: workPackageNumber,
        startDate,
        endDate,
      },
    },
    location,
  } = useTenantStore.getState().getAllEntities();

  const locationsFilter: Record<string, string> = {};

  const locationsByProject = descriptions?.projects?.find(
    ({ id }) => id === filters.projectId
  );

  if (locationsByProject) {
    locationsFilter[workPackageName.label] = locationsByProject.name;

    // Include the project number
    // TODO: check if this is going to be updated with externalKey
    if (locationsByProject?.number) {
      locationsFilter[workPackageNumber.label] = locationsByProject.number;
    }

    const selectedLocations = extractNamesBySelectedFilters(
      filters.locationIds,
      locationsByProject.locations
    );

    if (selectedLocations) {
      locationsFilter[location.labelPlural] = selectedLocations;
    }
  }

  return [
    {
      [startDate.label]: filters?.startDate,
      [endDate.label]: filters?.endDate,
      ...locationsFilter,
    },
  ];
};

/** Description of the selected filters from tab to tab. */
export const useFiltersDescriptions = ({
  filters = undefined,
  tab,
  descriptions = {},
}: FiltersDescriptionsParams): FiltersDescriptionsReturn => {
  if (!filters) {
    return [];
  }

  return tab === InsightsTab.PROJECT
    ? getProjectFilters({ filters, descriptions } as GetProjectFiltersArgs)
    : getPortfolioFilters({ filters, descriptions } as GetPortfolioFiltersArgs);
};
