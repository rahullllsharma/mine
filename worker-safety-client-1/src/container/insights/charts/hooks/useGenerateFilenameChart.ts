import type { FiltersDescriptionsReturn } from "./useFiltersDescriptions";
import {
  generateExportedProjectFilename,
  generateExportedPortfolioFilename,
} from "@/utils/files/shared";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { InsightsTab } from "../types";

export type UseGenerateFilenameChartOptions = {
  filters?: FiltersDescriptionsReturn;
};

/**
 * Generates the file name for the chart by tab and chart name
 */
export const useGenerateFilenameChart = (
  tab: InsightsTab,
  title: string,
  options?: UseGenerateFilenameChartOptions
): string => {
  const {
    getAllEntities,
    tenant: { name: tenantName },
  } = useTenantStore(state => state);

  if (tab === InsightsTab.PORTFOLIO) {
    return generateExportedPortfolioFilename({
      tenant: tenantName,
      title,
    });
  }

  const { filters = [] } = options || {};
  const {
    workPackage: {
      attributes: { name, externalKey },
    },
  } = getAllEntities();

  return generateExportedProjectFilename({
    title,
    project: {
      name: (filters?.[0]?.[name.label] || "") as string,
      number: (filters?.[0]?.[externalKey.label] || 0) as string,
    },
  });
};
