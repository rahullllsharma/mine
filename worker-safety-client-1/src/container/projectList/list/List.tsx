import type { Project } from "@/types/project/Project";
import type { LoaderParams as ListInfiniteLoaderParams } from "./components/loader/Loader";
import { useIsFirstRender } from "usehooks-ts";
import ProjectTable from "@/container/table/ProjectTable";
import { ProjectItems } from "./components/projectItems/ProjectItems";
import { Loader as ListInfiniteLoader } from "./components/loader/Loader";

type ListState = "loading" | "complete";

type ListParams = {
  projects: Project[];
  state?: ListState;
} & Partial<Pick<ListInfiniteLoaderParams, "onLoadMore">>;

function List({ projects, state, onLoadMore }: ListParams) {
  const isFirstRender = useIsFirstRender();
  const isLoading = state === "loading";
  const isComplete = state === "complete";

  const hasInfiniteScroll = !!(onLoadMore && !isComplete);

  return (
    <div className="overflow-y-auto">
      {/* desktop */}
      <div className="hidden lg:flex">
        <ProjectTable projects={projects} />
      </div>
      {/* mobile */}
      <div className="lg:hidden grid gap-4 grid-cols-auto-fill-list-card pb-16">
        <ProjectItems projects={projects} />
      </div>
      {/* Include infinite scroll support */}
      {hasInfiniteScroll && (
        <ListInfiniteLoader
          onLoadMore={onLoadMore}
          isLoading={isLoading}
          shouldLoadMore={!isFirstRender}
        />
      )}
    </div>
  );
}

export { List };
export type { ListState, ListParams };
