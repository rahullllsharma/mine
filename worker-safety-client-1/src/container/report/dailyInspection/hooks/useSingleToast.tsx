import type { ToastType } from "@/components/shared/toast/toastItem/Toast";

import type { ToastItem } from "@/components/shared/toast/context/ToastContext";
import { useContext } from "react";
import ToastContext from "@/components/shared/toast/context/ToastContext";

type ReplaceActiveToast = (type: ToastType, message: string) => void;

/**
 * Only display one toast at each time.
 *
 * @description
 * Pop the last toast and replace with a new one.
 *
 * @returns [(type: ToastType, message: string) => void]
 *
 * @deprecated DO NOT USE THIS, will be removed
 */
const useSingleToast = (): [ReplaceActiveToast] => {
  let lastActiveToast: ToastItem;
  const toastCtx = useContext(ToastContext);

  const replaceActiveToast: ReplaceActiveToast = (type, message) => {
    if (!lastActiveToast) {
      [lastActiveToast] = toastCtx?.items || [];
    }

    if (toastCtx?.items.some(item => item.id === lastActiveToast?.id)) {
      toastCtx?.dismissToast(lastActiveToast.id);
    }

    toastCtx?.pushToast(type, message);
    [lastActiveToast] = toastCtx?.items || [];
  };

  return [replaceActiveToast];
};

export default useSingleToast;
