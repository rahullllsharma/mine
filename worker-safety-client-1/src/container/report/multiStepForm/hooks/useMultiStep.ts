import type {
  MultiStepFormActions,
  MultiStepFormState,
  MultiStepFormStateItem,
} from "../state/reducer";
import type { Dispatch } from "react";
import type { NavigationStatus } from "@/components/navigation/Navigation";
import { useFormContext } from "react-hook-form";
import { noop } from "lodash-es";
import { useMultiStepFormContext } from "../context/MultiStepFormContext";

const useSafeDispatchMove = (dispatch: Dispatch<MultiStepFormActions>) => {
  const methods = useFormContext();

  if (!methods) {
    return (formActions: MultiStepFormActions) => dispatch(formActions);
  }

  const {
    formState: { isDirty, isSubmitted },
  } = methods;

  const canMoveWithoutConfirmation = !isDirty || isSubmitted;

  if (canMoveWithoutConfirmation) {
    return (formActions: MultiStepFormActions) => dispatch(formActions);
  }

  return (formActions: MultiStepFormActions) =>
    confirm("Discard unsaved changes?") ? dispatch(formActions) : noop();
};

type UseMultiStepActions = {
  moveTo: (section: string) => void;
  moveForward: () => void;
  moveBack: () => void;
  moveForwardAndComplete: () => void;
  markCurrentAs: (status: NavigationStatus) => void;
  markSectionsAs: (status: NavigationStatus, sections: string[]) => void;
};

type UseMultiStepState = {
  steps: MultiStepFormState;
  current: MultiStepFormStateItem;
  areAllStepsCompleted: boolean;
};

export const useMultiStepActions = (): UseMultiStepActions => {
  const [state, dispatch] = useMultiStepFormContext();
  const dispatchMove = useSafeDispatchMove(dispatch);

  return {
    moveTo: (section: string) =>
      dispatchMove({
        type: "JUMP_TO",
        value: section,
      }),
    moveForward: () =>
      dispatchMove({
        type: "NEXT",
      }),
    moveBack: () =>
      dispatchMove({
        type: "BACK",
      }),
    moveForwardAndComplete: () =>
      dispatchMove({
        type: "MOVE_FORWARD_AND_COMPLETE",
      }),
    markCurrentAs: (status: NavigationStatus) =>
      dispatch({
        type: "CHANGE_SECTIONS_STATUS",
        value: {
          status,
          sections: [
            (state.find(step => step.isSelected) as MultiStepFormStateItem).id,
          ],
        },
      }),
    markSectionsAs: (status: NavigationStatus, sections: string[]) =>
      dispatch({
        type: "CHANGE_SECTIONS_STATUS",
        value: { status, sections },
      }),
  };
};

export const useMultiStepState = (): UseMultiStepState => {
  const [state] = useMultiStepFormContext();
  return {
    steps: state,
    current: state.find(step => step.isSelected) as MultiStepFormStateItem,
    areAllStepsCompleted: state.every(step => step.status === "completed"),
  };
};
