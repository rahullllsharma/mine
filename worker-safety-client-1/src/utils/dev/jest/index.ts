export * from "./intl.utils";
export * from "./testing.utils";
export * from "./charts.utils";
