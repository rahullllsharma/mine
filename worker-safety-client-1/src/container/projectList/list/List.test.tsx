import { render, screen } from "@testing-library/react";
import { mockIntersectionObserver } from "jsdom-testing-mocks";
import { useProjects } from "@/hooks/useProjects";
import { mockTenantStore } from "@/utils/dev/jest";
import { List as ProjectsList } from "./List";

// eslint-disable-next-line react-hooks/rules-of-hooks
const projects = useProjects();
mockIntersectionObserver();

describe(ProjectsList.name, () => {
  beforeAll(() => {
    mockTenantStore();
  });

  it("should render correctly", () => {
    const { asFragment } = render(<ProjectsList projects={projects} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe("<ListInfiniteLoader />", () => {
    it("should render the component when onLoadMore callback is defined and it can still load more data", () => {
      render(<ProjectsList projects={projects} onLoadMore={jest.fn()} />);
      expect(screen.getByTestId("list-project-loader")).toBeInTheDocument();
    });

    it("should not render the component when onLoadMore callback is not defined", async () => {
      render(<ProjectsList projects={projects} />);
      expect(
        await screen.queryByTestId("list-project-loader")
      ).not.toBeInTheDocument();
    });

    it("should remove component when there is nothing more to fetch", async () => {
      render(
        <ProjectsList
          projects={projects}
          state="complete"
          onLoadMore={jest.fn()}
        />
      );

      expect(
        screen.queryByTestId("list-project-loader")
      ).not.toBeInTheDocument();
    });

    it("should remove the component after loading all data available", async () => {
      const { rerender } = render(
        <ProjectsList projects={projects} onLoadMore={jest.fn()} />
      );

      expect(
        await screen.findByTestId("list-project-loader")
      ).toBeInTheDocument();

      rerender(
        <ProjectsList
          projects={projects}
          state="loading"
          onLoadMore={jest.fn()}
        />
      );
      expect(
        await screen.findByTestId("list-project-loader")
      ).toBeInTheDocument();

      rerender(
        <ProjectsList
          projects={projects}
          state="complete"
          onLoadMore={jest.fn()}
        />
      );

      expect(
        await screen.queryByTestId("list-project-loader")
      ).not.toBeInTheDocument();
    });
  });
});
