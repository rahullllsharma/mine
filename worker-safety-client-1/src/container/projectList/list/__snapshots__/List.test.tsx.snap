// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
<DocumentFragment>
  <div
    class="overflow-y-auto"
  >
    <div
      class="hidden lg:flex"
    >
      <div
        class="w-full flex bg-white"
      >
        <div
          class="flex-1 flex flex-col overflow-y-auto"
          role="table"
          style="min-width: 0px;"
        >
          <div
            class="_thead bg-white z-10 border-b border-gray-200 divide-y divide-brand-gray-20 sticky top-0 px-4"
            data-testid="table-header"
          >
            <div
              class="_tr justify-between"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-workPackageName"
                role="columnheader"
                style="box-sizing: border-box; flex: 180 0 auto; min-width: 0px; width: 180px;"
              >
                Project Name
              </div>
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-todayRisk"
                role="columnheader"
                style="box-sizing: border-box; flex: 140 0 auto; min-width: 0px; width: 140px;"
              >
                Todays risk
              </div>
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-supervisor"
                role="columnheader"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                Supervisor
              </div>
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-libraryRegion"
                role="columnheader"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                Region
              </div>
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-libraryProjectType"
                role="columnheader"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px;"
              >
                Project Type
              </div>
              <div
                class="_th pt-5 pb-3 font-bold uppercase text-neutral-shade-75 truncate"
                colspan="1"
                data-testid="table-header-libraryDivision"
                role="columnheader"
                style="box-sizing: border-box; flex: 60 0 auto; min-width: 0px; width: 60px;"
              >
                Division
              </div>
            </div>
          </div>
          <div
            class="_tbody flex flex-col gap-2.5"
            data-testid="table-body"
            role="rowgroup"
          >
            <div
              class="_tr px-4 py-1 min-h-[3.5rem] bg-white  flex items-center"
              id="0"
              index="0"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 180 0 auto; min-width: 0px; width: 180px;"
              >
                <a
                  class="font-medium text-brand-urbint-50 leading-5 cursor-pointer inline-flex items-center active:text-brand-urbint-60 hover:text-brand-urbint-40"
                  href="/projects/7114445f-966a-42e2-aa3b-0e4d5365075d"
                >
                  <span
                    class=""
                  >
                    5th Street Main Relocation
                  </span>
                </a>
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 140 0 auto; min-width: 0px; width: 140px;"
              >
                <span
                  class="inline-flex items-center"
                >
                  <span
                    class="inline-flex items-center w-full"
                  >
                    <div
                      class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-high text-white"
                      role="note"
                    >
                      <span
                        class="flex items-center px-1"
                      >
                        <i
                          aria-hidden="true"
                          class="ci-chevron_up"
                        />
                        <span
                          class="text-tiny mr-1"
                        >
                          HIGH
                        </span>
                      </span>
                    </div>
                  </span>
                </span>
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                Jakob Aminoff
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                NE (New England)
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px;"
              >
                Main replacement
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 60 0 auto; min-width: 0px; width: 60px;"
              >
                Gas
              </div>
            </div>
            <div
              class="_tr px-4 py-1 min-h-[3.5rem] bg-white  flex items-center"
              id="1"
              index="1"
              role="row"
              style="display: flex; flex: 1 0 auto; min-width: 0px;"
            >
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 180 0 auto; min-width: 0px; width: 180px;"
              >
                <a
                  class="font-medium text-brand-urbint-50 leading-5 cursor-pointer inline-flex items-center active:text-brand-urbint-60 hover:text-brand-urbint-40"
                  href="/projects/7114445f-966a-42e2-aa3b-0e4d53650751a"
                >
                  <span
                    class=""
                  >
                    N. Washington Street Bridge
                  </span>
                </a>
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 140 0 auto; min-width: 0px; width: 140px;"
              >
                <span
                  class="inline-flex items-center"
                >
                  <span
                    class="inline-flex items-center w-full"
                  >
                    <div
                      class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-low text-white"
                      role="note"
                    >
                      <span
                        class="flex items-center px-1"
                      >
                        <i
                          aria-hidden="true"
                          class="ci-chevron_down"
                        />
                        <span
                          class="text-tiny mr-1"
                        >
                          LOW
                        </span>
                      </span>
                    </div>
                  </span>
                </span>
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                Jakob Aminoff
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 120 0 auto; min-width: 0px; width: 120px;"
              >
                NE (New England)
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 100 0 auto; min-width: 0px; width: 100px;"
              >
                Bridge replacement
              </div>
              <div
                class="_td text-sm pr-4 last:pr-0"
                role="cell"
                style="box-sizing: border-box; flex: 60 0 auto; min-width: 0px; width: 60px;"
              >
                Gas
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="lg:hidden grid gap-4 grid-cols-auto-fill-list-card pb-16"
    >
      <div
        class="flex rounded bg-white shadow-5"
      >
        <button
          class="flex-1 p-6 text-left"
        >
          <header
            class="text-neutral-shade-100 text-lg font-semibold"
          >
            5th Street Main Relocation
          </header>
          <div
            class="mt-4"
          >
            <div
              class="flex text-sm items-center py-1.5"
            >
              <span
                class="flex-1 text-neutral-shade-100 text-sm font-semibold"
              >
                Project Risk
              </span>
              <span
                class="inline-flex items-center"
              >
                <span
                  class="inline-flex items-center w-full"
                >
                  <div
                    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-high text-white"
                    role="note"
                  >
                    <span
                      class="flex items-center px-1"
                    >
                      <i
                        aria-hidden="true"
                        class="ci-chevron_up"
                      />
                      <span
                        class="text-tiny mr-1"
                      >
                        HIGH
                      </span>
                    </span>
                  </div>
                </span>
              </span>
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Region
              </span>
              NE (New England)
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Supervisor
              </span>
              Jakob Aminoff
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Project Type
              </span>
              Main replacement
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Division
              </span>
              Gas
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex rounded bg-white shadow-5"
      >
        <button
          class="flex-1 p-6 text-left"
        >
          <header
            class="text-neutral-shade-100 text-lg font-semibold"
          >
            N. Washington Street Bridge
          </header>
          <div
            class="mt-4"
          >
            <div
              class="flex text-sm items-center py-1.5"
            >
              <span
                class="flex-1 text-neutral-shade-100 text-sm font-semibold"
              >
                Project Risk
              </span>
              <span
                class="inline-flex items-center"
              >
                <span
                  class="inline-flex items-center w-full"
                >
                  <div
                    class="flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 whitespace-nowrap bg-risk-low text-white"
                    role="note"
                  >
                    <span
                      class="flex items-center px-1"
                    >
                      <i
                        aria-hidden="true"
                        class="ci-chevron_down"
                      />
                      <span
                        class="text-tiny mr-1"
                      >
                        LOW
                      </span>
                    </span>
                  </div>
                </span>
              </span>
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Region
              </span>
              NE (New England)
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Supervisor
              </span>
              Jakob Aminoff
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Project Type
              </span>
              Bridge replacement
            </div>
            <div
              class="flex items-center border-solid border-t py-1.5 text-sm text-neutral-shade-75"
            >
              <span
                class="flex-1 font-semibold"
              >
                Division
              </span>
              Gas
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
